docs/adr/0039-locales-languages.md

```diff
@@ -0,0 +1,183 @@
+
+### Format BCP 47, identifiants de locales
+
+La syntaxe des *languages tags* est puissante et extensible, pour les détails on se reportera aux spécifications citées.
+
+On considérera initialement les identifiants de locales suivants pour pix. Cette liste qui correspond aux locales actuellement nécessaires pour pix a vocation à s'allonger :
+
+* `fr`
+* `fr-FR`
+* `fr-BE`
+* `en`
+
+Et ce standard permet également de gérer des publics vivant dans une région mais ne maîtrisant pas la langue de la région, ou souhaitant utiliser une autre langue, par exemple :
+
+* `en-FR`
+* `en-BE`
+ 
+[`Intl.Locale`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale) est un builtin JavaScript qui permet de manipuler les locales comme suit :
+
+```
+> locale = new Intl.Locale("fr-FR")
+> locale.toString()
+'fr-FR'
+> locale.language
+'fr'
+> locale.region
+'FR'
+
+> locale = new Intl.Locale("fr-BE")
+> locale.toString()
+'fr-BE'
+> locale.language
+'fr'
+> locale.region
+'BE'
+
+> locale = new Intl.Locale("fr")
+> locale.toString()
+'fr'
+> locale.language
+'fr'
+> locale.region
+undefined
+
+> locale = new Intl.Locale("en")
+> locale.toString()
+'en'
+> locale.language
+'en'
+> locale.region
+undefined
+```
+
+La spécification *BCP 47* accepte que la `region` soit écrite en minuscules, mais la forme canonique de la `region` est en majuscules. Le builtin `Intl.Locale` suit bien cette recommandation et on peut s'en rendre compte en instanciant un nouvel objet en utilisant un identifiant dans une forme non recommandée comme `fr-fr` qui sera transformée dans la forme canonique `fr-FR` :
+
+```
+> locale = new Intl.Locale("fr-fr")
+> locale.toString()
+'fr-FR'
+> locale.language
+'fr'
+> locale.region
+'FR'
+```
+
+#### Monnaie (currency)
+
+Alors que par définition les paramètres régionaux contiennent la monnaie, le builtin [`Intl.Locale`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale) ne contient rien sur le système de monnaie (currency) d'une région. A priori ce n'est pas un besoin pour Pix, mais on le mentionne ici car il faudrait creuser pour comprendre le pourquoi de cette exception, pour ne pas risquer de manquer des notions/détails potentiellement importants.
+
+### État du code de Pix
+
+Actuellement dans le code de pix les 2 notions et noms de *locale* et *language* sont manipulés :
+
+* la notion et le nom de *language* sont utilisés pour le côté client
+* la notion et le nom de *locale* sont utilisés côté BDD et hapi
+
+En ce qui concerne le format des identifiants de *language*/*locale*, c'est le format suivant qui est utilisé dans le code de pix : `fr`, `fr-fr`, `fr-be`. Ce format n'est pas la forme canonique de représentation tel que défini par la spécification *BCP 47* (car la partie `region` devrait être en majuscules), mais c'est une forme acceptée et par spécification c'est compatible avec `Intl.Locale`.
```

Comment:
A noter que coté site vitrine, les informations proviennent de Prismic. Hors, sur Prismic, les locales sont notés en minuscule (`fr-fr`) et non en BCP 47.

Reply:
Je vais noter d'investiguer ce point. Merci !

Reply:
Je viens de tester de faire un fetch sur Prismic en spécifiant `fr-FR` au lieu de `fr-fr` et `fr-BE` au lieu de `fr-be` et Prismic renvoie les mêmes résultats. Cela veut dire que même si Prismic n'utilise pas la casse recommandée par PBC47, apparemment ils pourraient en suivre les règles d'égalité.

Reply:
Indépendamment de cette petite impédance au niveau de la casse des identifiants de locales, compte tenu que du renommage de locales va avoir lieu (`en-gb`→`en`) un _anticorruption layer_ (DDD) semble de toute façon nécessaire.